tests/benchmarks/test_model_schema_generation.py

```diff
@@ -170,163 +178,44 @@ class Product(BaseModel):
-def test_custom_field_validator_via_decorator(benchmark, validator_mode) -> None:
-    def schema_gen() -> None:
-        class ModelWithFieldValidator(BaseModel):
-            field: Any
-
-            @field_validator('field', mode=validator_mode)
-            @classmethod
-            def validate_field(cls, v: Any):
-                return v
-
-    benchmark(schema_gen)
-
-
-@pytest.mark.benchmark(group='model_schema_generation')
-def test_custom_wrap_field_validator_via_decorator(benchmark) -> None:
-    def schema_gen() -> None:
-        class ModelWithWrapFieldValidator(BaseModel):
-            field: Any
-
-            @field_validator('field', mode='wrap')
-            @classmethod
-            def validate_field(cls, v: Any, handler: ValidatorFunctionWrapHandler) -> Any:
-                return handler(v)
-
-    benchmark(schema_gen)
-
-
-@pytest.mark.parametrize('validator_constructor', [BeforeValidator, AfterValidator, PlainValidator])
-@pytest.mark.benchmark(group='model_schema_generation')
-def test_custom_field_validator_via_annotation(benchmark, validator_constructor) -> None:
-    def validate_field(v: Any) -> Any:
-        return v
-
-    def schema_gen(validation_func) -> None:
-        class ModelWithFieldValidator(BaseModel):
-            field: Annotated[Any, validator_constructor(validation_func)]
-
-    benchmark(schema_gen, validate_field)
-
-
-@pytest.mark.benchmark(group='model_schema_generation')
-def test_custom_wrap_field_validator_via_annotation(benchmark) -> None:
-    def validate_field(v: Any, handler: ValidatorFunctionWrapHandler) -> Any:
-        return handler(v)
-
-    def schema_gen(validator_func: Callable) -> None:
-        class ModelWithWrapFieldValidator(BaseModel):
-            field: Annotated[Any, WrapValidator(validator_func)]
-
-    benchmark(schema_gen, validate_field)
-
+        models.append(create_model(model_name, __config__={'defer_build': True}, **model_fields))
 
-@pytest.mark.benchmark(group='model_schema_generation')
-def test_custom_model_validator_before(benchmark):
-    def schema_gen() -> None:
-        class ModelWithBeforeValidator(BaseModel):
-            field: Any
-
-            @model_validator(mode='before')
-            @classmethod
-            def validate_model_before(cls, data: Any) -> Any:
-                return data
+    def rebuild_models(models: List[Type[BaseModel]]) -> None:
+        for model in models:
+            rebuild_model(model)
 
-    benchmark(schema_gen)
+    benchmark(rebuild_models, models)
 
 
 @pytest.mark.benchmark(group='model_schema_generation')
-def test_custom_model_validator_after(benchmark) -> None:
-    def schema_gen() -> None:
-        class ModelWithAfterValidator(BaseModel):
-            field: Any
-
-            @model_validator(mode='after')
-            def validate_model_after(self: 'ModelWithAfterValidator') -> 'ModelWithAfterValidator':
-                return self
+def test_field_validators_serializers(benchmark) -> None:
```

Comment:
I condensed everything in one benchmark because:
- We don't expect decorators vs Annotated metadata classes to make a difference, converting from the decorator to the `*Validator/*Serializer` class probably takes a few ns at most.
- We don't expect any changes in performance between modes, so there's no need to make them separate benchmarks

(code coverage is still untouched with these changes)